File: app/src/main/java/com/example/memodown/AndroidUtil.java
package com.example.memodown;

import android.content.res.Resources;
import android.util.DisplayMetrics;


public class AndroidUtil
{

    public static float pixels2Dp(float px)
    {
        DisplayMetrics metrics=Resources.getSystem().getDisplayMetrics();
        float dp=px/(metrics.densityDpi/160f);
        return Math.round(dp);
    }

    public static float dp2Pixels(float dp)
    {
        DisplayMetrics metrics=Resources.getSystem().getDisplayMetrics();
        float px=dp*(metrics.densityDpi/160f);
        return Math.round(px);
    }


    private int convertDpToPx(int dp)
    {
        return Math.round(dp*(Resources.getSystem().getDisplayMetrics().xdpi/DisplayMetrics.DENSITY_DEFAULT));

    }

    private int convertPxToDp(int px)
    {
        return Math.round(px/(Resources.getSystem().getDisplayMetrics().xdpi/DisplayMetrics.DENSITY_DEFAULT));
    }

}

File: app/src/main/java/com/example/memodown/AndroidUtil.kt
package com.example.memodown

import android.content.res.Resources
import android.util.DisplayMetrics
import kotlin.math.roundToInt

public class AndroidUtil{
    fun pixels2Dp(px: Float): Float {
        val metrics = Resources.getSystem().displayMetrics
        val dp: Float = px / (metrics.densityDpi/160f)
        return dp.roundToInt().toFloat()
    }

    fun dp2Pixels(dp: Float): Float{
        val metrics = Resources.getSystem().displayMetrics
        val px: Float = dp * (metrics.densityDpi/160f)
        return px.roundToInt().toFloat()
    }

    private fun convertDpTopx(dp: Int): Int{
        return (dp*(Resources.getSystem().displayMetrics.xdpi / DisplayMetrics.DENSITY_DEFAULT)).roundToInt()
    }

    private fun convertPxToDp(px: Int): Int{
        return (px/(Resources.getSystem().displayMetrics.xdpi / DisplayMetrics.DENSITY_DEFAULT)).roundToInt()
    }
}